solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudSolrClient.java

```diff
@@ -1001,25 +999,30 @@ protected NamedList<Object> sendRequest(SolrRequest<?> request, List<String> inp
 
     boolean sendToLeaders = false;
 
-    if (request instanceof IsUpdateRequest) {
-      sendToLeaders = ((IsUpdateRequest) request).isSendToLeaders() && this.isUpdatesToLeaders();
+    if (request.getRequestType() == SolrRequestType.UPDATE) {
```

Comment:
this check seems pointless since we're about to see if request is an instanceof UpdateRequest